packages/core/src/agent/MessageSender.ts

```diff
@@ -250,7 +275,14 @@ export class MessageSender {
     }
 
     const outboundPackage = await this.packMessage({ message, keys, endpoint: service.serviceEndpoint })
-    await this.outboundTransporter.sendMessage(outboundPackage)
+    outboundPackage.endpoint = service.serviceEndpoint
+    const protocol = outboundPackage.endpoint.split(':')[0]
```

Comment:
Maybe we can add `protocolScheme` getter to the service class?

so we can do 
```
transport.supportedSchemes.includes(service.protocolScheme)
```

Reply:
I would use a `protocolScheme` getter if the service class had it. We should add it.

Reply:
Awesome! Could you add it as part of this PR?